scanner: Shuffle disks to scan

vadmeste opened the pull request:
## Description
Ensure random association between disk and bucket in each crawling
iteration to ensure that ILM applies correctly to objects no present in
all disks.

## Motivation and Context
Better scanning

## How to test this PR?
Enable MINIO_DISK_USAGE_SCANNER_DEBUG=on and see 
what disks are being scanned

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

harshavardhana commented on cmd/erasure.go:
```diff
@@ -435,6 +435,11 @@ func (er erasureObjects) nsScanner(ctx context.Context, buckets []BucketInfo, bf
 
+	// Shuffle disks to ensure a total randomness of bucket/disk association to ensure
+	// that objects that are not present in all disks are accounted and ILM applied.
+	rand.Seed(time.Now().UnixNano())
```
It is better to use rand.New() and then use the resultant logic to shuffle.

klauspost reviewed (approved):
Yeah, don't use the global rng. Otherwise LGTM.

Commit:
scanner: Shuffle disks to scan

Ensure random association between disk and bucket in each crawling
iteration to ensure that ILM applies correctly to objects no present in
all disks.
